### PR TITLE
Improve safety warning on ptr::swap

### PR DESCRIPTION
**Pull request by steveklabnik**
r? @eddyb @bluss , who I asked about this on IRC

### CODE REVIEW

File: src/libcore/ptr.rs [hunk 1]
```diff
@@ -74,7 +74,10 @@ pub const fn null_mut<T>() -> *mut T { 0 as *mut T }
 ///
 /// # Safety
 ///
-/// This is only unsafe because it accepts a raw pointer.
+/// This function copies the memory through the raw pointers passed to it
+/// as arguments, which means they will be dereferenced.
```
Comment (GuillaumeGomez) [inline src/libcore/ptr.rs]: I don't think the warning is clear enough. I'd state something like:

"If the pointer is null or points to invalid data, the program will crash or have an undefined behaviour."

It seems very important to me to indicate the consequences.

**Review by GuillaumeGomez — CHANGES_REQUESTED** [review]

File: src/libcore/ptr.rs (hunk 1 above)
Comment (steveklabnik) [inline src/libcore/ptr.rs]: This is already defined at the language level; repeating it here feels extraneous.

Comment (bluss) [inline src/libcore/ptr.rs]: To say that it reads and writes through the raw pointers should be enough.

**Comment by steveklabnik** [comment]
@bluss updated

**Review by eddyb — APPROVED** [review]
LGTM. r=me if @bluss is satisfied.

**Review by GuillaumeGomez — APPROVED** [review]

**Pull request merged**